thirdparty/stdatomic/win/atomic.h

```diff
@@ -315,8 +315,13 @@ static inline unsigned short atomic_load_u16(unsigned short* object) {
 static inline int atomic_compare_exchange_strong_u16(unsigned short* object, unsigned short* expected,
                                                  unsigned short desired)
 {
-	unsigned short old = *expected;
+    unsigned short old = *expected;
+// TODO: on tcc+win32 shared_lock_test.v, e.g., can be compiled but "cannot find entry point of InterlockedCompareExchange16" at runtime
+#if defined(__TINYC__) && defined(WIN32)
+    *expected = InterlockedCompareExchange(object, desired, old);
```

Comment:
I don't think this is correct. `InterlockedCompareExchange()` operates on 32 bit values and `atomic_compare_exchange_strong_u16()` is supposed to operate on 16 bit values. Why is this case distinction necessary? Can't we use `InterlockedCompareExchange16()` in all cases?

Reply:
As said, because with tcc backend on win32 platform I got error "cannot find entry point of InterlockedCompareExchange16" at runtime, although compilation is nice.

Reply:
Hmm, `InterlockedCompareExchange16()` is supposed to be a compiler intrinsic but in tcc's `winnt.h` there is only an inline assembler implementation for 64 bit architectures. Maybe a version for 32 bit can be added there.
Another possibility would be a kind of emulation using `InterlockedCompareExchange()` that takes alignment into account and makes sure that the other half of the 32 bit value is unchanged. This would probably be possible but the inline assembler approach is easier and surely preferable.

Reply:
Your suggestion is very helpful. Just solved it by added a 32-bit implementation simply through copying 64-bit's.